ftw/util.py

```diff
@@ -98,3 +97,22 @@ def extract_yaml(yaml_files):
             print('General error')
             raise e
     return loaded_yaml
+    
+    
+def ensure_str(s, encoding='utf-8', errors='strict'):
+    # Optimization: Fast return for the common case.
+    if type(s) is str:
+        return s
+    if isinstance(s, bytes):
+        return s.decode(encoding, errors)
+    elif not isinstance(s, (str, bytes)):
+        raise TypeError("not expecting type '%s'" % type(s))
+
+
+def ensure_binary(s, encoding='utf-8', errors='strict'):
+    if isinstance(s, bytes):
+        return s
+    if isinstance(s, str):
+        return s.encode(encoding, errors)
+    raise TypeError("not expecting type '%s'" % type(s))
+
```

Comment:
We can leave these for now but I think we want to ensure that the data is the right type so we don't need these functions.

Reply:
ack.